pkg/sbom/io/decode.go

```diff
@@ -0,0 +1,340 @@
+package io
+
+import (
+	"errors"
+	"sort"
+	"strconv"
+
+	"github.com/package-url/packageurl-go"
+	"go.uber.org/zap"
+	"golang.org/x/exp/maps"
+	"golang.org/x/xerrors"
+
+	ftypes "github.com/aquasecurity/trivy/pkg/fanal/types"
+	"github.com/aquasecurity/trivy/pkg/log"
+	"github.com/aquasecurity/trivy/pkg/purl"
+	"github.com/aquasecurity/trivy/pkg/sbom/core"
+	"github.com/aquasecurity/trivy/pkg/types"
+	"github.com/aquasecurity/trivy/pkg/uuid"
+)
+
+var (
+	ErrPURLEmpty       = errors.New("purl empty error")
+	ErrUnsupportedType = errors.New("unsupported type")
+)
+
+type Decoder struct {
+	bom *core.BOM
+
+	osID uuid.UUID
+	pkgs map[uuid.UUID]*ftypes.Package
+	apps map[uuid.UUID]*ftypes.Application
+}
+
+func NewDecoder(bom *core.BOM) *Decoder {
+	return &Decoder{
+		bom:  bom,
+		pkgs: make(map[uuid.UUID]*ftypes.Package),
+		apps: make(map[uuid.UUID]*ftypes.Application),
+	}
+}
+
+func (m *Decoder) Decode(sbom *types.SBOM) error {
+	// Parse the root component
+	if err := m.decodeRoot(sbom); err != nil {
+		return xerrors.Errorf("failed to decode root component: %w", err)
+	}
+
+	// Parse all components
+	if err := m.decodeComponents(sbom); err != nil {
+		return xerrors.Errorf("failed to decode components: %w", err)
+	}
+
+	// Build dependency graph between packages
+	m.buildDependencyGraph()
+
+	// Add OS packages
+	m.addOSPkgs(sbom)
+
+	// Add language-specific packages
+	m.addLangPkgs(sbom)
+
+	// Add remaining packages
+	if err := m.addOrphanPkgs(sbom); err != nil {
+		return xerrors.Errorf("failed to aggregate packages: %w", err)
+	}
+
+	sort.Slice(sbom.Applications, func(i, j int) bool {
+		if sbom.Applications[i].Type != sbom.Applications[j].Type {
+			return sbom.Applications[i].Type < sbom.Applications[j].Type
+		}
+		return sbom.Applications[i].FilePath < sbom.Applications[j].FilePath
+	})
+
+	sbom.BOM = m.bom
+
+	return nil
+}
+
+func (m *Decoder) decodeRoot(s *types.SBOM) error {
+	root := m.bom.Root()
+	if root == nil {
+		return nil // No root found
+	}
+
+	var err error
+	for _, prop := range root.Properties {
+		switch prop.Name {
+		case core.PropertyImageID:
+			s.Metadata.ImageID = prop.Value
+		case core.PropertySize:
+			s.Metadata.Size, err = strconv.ParseInt(prop.Value, 10, 64)
+			if err != nil {
+				return xerrors.Errorf("failed to convert size: %w", err)
+			}
```

Comment:
nit:
```suggestion
			if s.Metadata.Size, err = strconv.ParseInt(prop.Value, 10, 64);  err != nil {
				return xerrors.Errorf("failed to convert size: %w", err)
			}
```

Reply:
fixed in https://github.com/aquasecurity/trivy/pull/6240/commits/d3de909fa6eb8fba4f501fafdd98c8aefb8531d2